### PR TITLE
[Merged by Bors] - feat: trace of a permutation matrix

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
To go with Matrix.det_permutation

---

Question: this adds several imports to state the theorem. Should it be put elsewhere?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dupuisf** [comment]
> Question: this adds several imports to state the theorem. Should it be put elsewhere?

The safe answer would be yes!

**Comment by Ruben-VandeVelde** [comment]
Okay, suggestions? :)

**Comment by dupuisf** [comment]
Does `#find_home` say anything interesting?

**Comment by Ruben-VandeVelde** [comment]
Nothing at all

**Comment by dupuisf** [comment]
New file then?

File: Mathlib/LinearAlgebra/Matrix/Trace.lean [hunk 1]
```diff
@@ -196,6 +199,11 @@ lemma trace_submatrix_succ {n : ℕ} [NonUnitalNonAssocSemiring R]
   rw [← (finSuccEquiv n).symm.sum_comp]
   simp
 
+theorem trace_permutation [DecidableEq n] [AddCommMonoidWithOne R] (σ : Equiv.Perm n) :
+    Matrix.trace (σ.toPEquiv.toMatrix : Matrix n n R) = (Function.fixedPoints σ).ncard := by
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Matrix/Trace.lean:203]: If we are going to start a new file, may I suggest an
```lean
abbrev Equiv.Perm.permMatrix (σ : Equiv.Perm n) : Matrix n n R := σ.toPEquiv.toMatrix
```
And then the existing
```lean
-- Mathlib/LinearAlgebra/Matrix/Determinant.lean
275:    Matrix.det (σ.toPEquiv.toMatrix : Matrix n n R) = Perm.sign σ := by
276:  rw [← Matrix.mul_one (σ.toPEquiv.toMatrix : Matrix n n R), PEquiv.toPEquiv_mul_matrix,

-- Mathlib/Data/Matrix/PEquiv.lean
97:    (M : Matrix m n α) : f.toPEquiv.toMatrix * M = M.submatrix f id := by
102:theorem mul_toPEquiv_toMatrix {m n α : Type*} [Fintype n] [DecidableEq n] [Semiring α] (f : n ≃ n)
103:    (M : Matrix m n α) : M * f.toPEquiv.toMatrix = M.submatrix id f.symm :=
107:#align pequiv.mul_to_pequiv_to_matrix PEquiv.mul_toPEquiv_toMatrix
143:    (Equiv.swap i j).toPEquiv.toMatrix =
174:theorem equiv_toPEquiv_toMatrix [DecidableEq n] [Zero α] [One α] (σ : Equiv n n) (i j : n) :
175:    σ.toPEquiv.toMatrix i j = (1 : Matrix n n α) (σ i) j :=
177:#align pequiv.equiv_to_pequiv_to_matrix PEquiv.equiv_toPEquiv_toMatrix
```
could also move to this new file.

Comment (Ruben-VandeVelde) [reply Mathlib/LinearAlgebra/Matrix/Trace.lean]: `mul_toPEquiv_toMatrix` and `equiv_toPEquiv_toMatrix` are about `Equiv.toPEquiv`, not `Equiv.Perm.toEquiv`. Moved `det`, crediting Anne for https://github.com/leanprover-community/mathlib/pull/1816

**Pull request closed without merging**